trie/trie.go

```diff
@@ -1026,14 +1050,73 @@ func (t *Trie) resolveShadowNode(epoch types.StateEpoch, cur node, prefix []byte
 			}
 		}
 		return nil
-	case valueNode, hashNode:
+	case valueNode, hashNode, nil:
 		// just skip
 		return nil
 	default:
 		return errors.New("resolveShadowNode unsupported node type")
 	}
 }
 
+func (t *Trie) ShadowHash() (*common.Hash, error) {
+	if t.root == nil {
+		return nil, nil
+	}
+	h := newHasher(true)
+	defer returnHasherToPool(h)
+	return t.shadowHash(t.root, h, nil, t.root.getEpoch())
```

Comment:
should we use `t.root.getEpoch()` or `t.currentEpoch`? or both should be the same?

Reply:
when insert/delete a node, it updates the node's epoch and shadowNode epochMap from `t.currentEpoch`, so I prefer to use the node's epoch to represent its epoch.